python/tvm/relay/backend/vm.py

```diff
@@ -198,20 +198,29 @@ def _update_target(self, target):
         target = target if target else tvm.target.Target.current()
         if target is None:
             raise ValueError("Target is not set in env or passed as argument.")
-        tgts = {}
-        if isinstance(target, (str, tvm.target.Target)):
-            dev_type = tvm.tir.IntImm("int32", tvm.nd.device(str(target)).device_type)
-            tgts[dev_type] = tvm.target.Target(target)
-        elif isinstance(target, dict):
-            for dev, tgt in target.items():
-                dev_type = tvm.tir.IntImm("int32", tvm.nd.device(dev).device_type)
-                tgts[dev_type] = tvm.target.Target(tgt)
-        else:
+
+        elif isinstance(target, str):
+            target = {target: target}
+
```

Comment:
remove this line

Reply:
I think this line should be present, in order to maintain previous behavior.  The `target` variable comes from either `VMCompiler.lower` or `VMCompiler.optimize`, both of which state that the `target` parameter can be a string, a `tvm.target.Target`, or a dictionary.  If this line were removed, then there would be a type error later on when `target` is accessed as a dictionary, if a `str` argument is passed in.

The modified version moves all the input-handling into a single location, making a dictionary regardless of the input value, so that the remainder of the function doesn't need to repeat the logic for how to set the output `tgts` dictionary.